creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -0,0 +1,143 @@
+    const logger = genericLogger.child(logContext)
+
+    try {
+      // Update aggregate sync data
+      const aggregateSyncKey = SyncHistoryAggregator.getAggregateSyncKey(CREATOR_NODE_ENDPOINT)
+      const existingAggregateSyncEntry = await redisClient.get(aggregateSyncKey)
+      if (!existingAggregateSyncEntry) {
+      // Init aggregate sync data
+        await redisClient.set(aggregateSyncKey,
+          JSON.stringify({
+            triggered: 0,
+            success: 0,
+            fail: 0
+          }),
+          'EX', // seconds -- Set the specified expire time, in seconds.
+          EXPIRATION
+        )
+      }
+
+      let currentAggregateData = await redisClient.get(aggregateSyncKey)
+      currentAggregateData = JSON.parse(currentAggregateData)
+      currentAggregateData[state] += 1
+      currentAggregateData[SYNC_STATES.triggered] += 1
+
+      // Get the existing TTL and update the key with it
+      let aggregateSyncKeyTTL = await SyncHistoryAggregator.getKeyTTL(aggregateSyncKey)
+      await redisClient.set(aggregateSyncKey,
+        JSON.stringify(currentAggregateData),
+        'EX',
+        aggregateSyncKeyTTL
+      )
+
+      // Update latest sync data
+      const latestSyncKey = SyncHistoryAggregator.getLatestSyncKey(CREATOR_NODE_ENDPOINT)
+      const existingLatestSyncEntry = await redisClient.get(latestSyncKey)
+      if (!existingLatestSyncEntry) {
+        // Init latest sync data
+        await redisClient.set(latestSyncKey,
+          JSON.stringify({
+            success: null,
+            fail: null
+          }),
+          'EX',
+          EXPIRATION
+        )
+      }
+
+      let currentLatestSyncData = await redisClient.get(latestSyncKey)
+      currentLatestSyncData = JSON.parse(currentLatestSyncData)
+      currentLatestSyncData[state] = timeOfEvent
+
+      // Get the existing TTL and update the key with it
+      let latestSyncKeyTTL = await SyncHistoryAggregator.getKeyTTL(latestSyncKey)
+      await redisClient.set(latestSyncKey,
+        JSON.stringify(currentLatestSyncData),
+        'EX',
+        latestSyncKeyTTL
+      )
+
+      logger.info(`SyncHistoryAggregator - Successfully tracked ${state} sync at ${timeOfEvent} from ${CREATOR_NODE_ENDPOINT}`)
+    } catch (e) {
+      // Only log error to not block any main thread
+      logger.error(`SyncHistoryAggregator - Failed to track ${state} sync at ${timeOfEvent} from ${CREATOR_NODE_ENDPOINT}: ${e.toString()}`)
+    }
+  }
+
+  static async getKeyTTL (key) {
+    const ttl = await redisClient.ttl(key)
+    return ttl || EXPIRATION
+  }
+
+  static async getAggregateSyncData () {
+    const aggregateSyncKey = SyncHistoryAggregator.getAggregateSyncKey(CREATOR_NODE_ENDPOINT)
+    const currentAggregateData = await redisClient.get(aggregateSyncKey)
+
+    // Structure: {triggered: <number>, success: <number>, fail: <number>}
+    return currentAggregateData ? JSON.parse(currentAggregateData) : {}
```

Comment:
JSON.parse should be try/catched in the event of invalid JSON, which throws an error